js/autotrack/touchables.js

```diff
@@ -29,44 +30,45 @@ export const autotrackPress = track => (eventType, componentThis, event) => {
   track('touch', autotrackProps);
 };
 
-const unsafePressHandler = (event, track, isLongPress) => {
-  const autocaptureProps = getBaseComponentPropsFromFiber(event._targetInst);
-
-  if (!autocaptureProps) {
-    // We're not capturing this interaction.
-    return;
+export const withHeapTouchableAutocapture = track => TouchableComponent => {
+  class HeapTouchableAutocapture extends React.Component {
+    render() {
+      const { forwardedRef, onPress, onLongPress, ...rest } = this.props;
+
+      return (
+        <TouchableComponent
+          ref={forwardedRef}
+          onPress={e => {
+            bailOnError(autotrackPress(track))('touchableHandlePress', this, e);
+
+            onPress && onPress(e);
+          }}
+          onLongPress={e => {
+            bailOnError(autotrackPress(track))(
+              'touchableHandleLongPress',
+              this,
+              e
+            );
+
+            onLongPress && onLongPress(e);
+          }}
+          {...rest}
+        >
+          {this.props.children}
+        </TouchableComponent>
+      );
+    }
   }
 
-  autocaptureProps.is_long_press = isLongPress;
-
-  track('touch', autocaptureProps);
-};
-
-const pressHandler = bailOnError(unsafePressHandler);
-
-// Wrap the config passed to the 'Pressability' constructor by wrapping the 'onPress' and 'onLongPress' properties in the config, and
-// keeping all other properties as-is.  See the config passed to the 'Pressability' constructor in
-// https://github.com/facebook/react-native/blob/a5151c2b5f6f03896eb7d9df873c5f61a706f055/Libraries/Components/Touchable/TouchableOpacity.js#L143-L186.
-export const wrapPressabilityConfig = track => pressabilityConfig => {
-  const newConfig = {
-    ...pressabilityConfig,
-  };
-
-  if (newConfig.onPress) {
-    newConfig.onPress = event => {
-      pressHandler(event, track, false);
+  HeapTouchableAutocapture.displayName = `withHeapTouchableAutocapture(${getComponentDisplayName(
+    TouchableComponent
+  )})`;
 
-      pressabilityConfig.onPress(event);
-    };
-  }
-
-  if (newConfig.onLongPress) {
-    newConfig.onLongPress = event => {
-      pressHandler(event, track, true);
+  const forwardRefHoc = React.forwardRef((props, ref) => {
```

Comment:
forwardRef essentially just passes the ref to the children right?

Reply:
Yep, it basically makes it such that you can forward the ref to some internal component (in this case, so you get the ref for the touchable component, and not the HOC wrapping the touchable component)